### DISCUSSION TITLE
Installing the preconfigured Eclipse Development Environment on WSL2

### DISCUSSION THREAD

**Discussion by fedejeanne** [eclipse-platform/eclipse.platform#1360] [Q&A] [answered]
Hello everyone,
I would like to install the pre-configured IDE using Oomph (the one mentioned under [Creating an Eclipse Development Environment](https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#creating-an-eclipse-development-environment)) in my Windows machine using WSL2.

I already have WSL2 up and running and I can start the Oomph installer, it looks like this:

![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/d0cc988b-0bd2-4879-ab3e-08ee879021ec)

The problem I'm facing is: I can't _drag & drop_ the link in the documentation to the installer ...

![image](https://github.com/eclipse-platform/eclipse.platform/assets/2205684/f3620910-36f2-46b0-bf8b-67e2f86b58df)

... so I would like to circumvent that by "doing it manually".

Any hints on how to proceed, even if it doesn't follow my original approach (_i.e._ no _drag & drop_) are appreciated!

**Comment by merks** [comment, marked as answer]
If you follow the link to here, there are instructions for various approaches.   😱 

https://www.eclipse.org/setups/installer/?url=https://raw.githubusercontent.com/eclipse-platform/eclipse.platform.releng.aggregator/master/oomph/PlatformSDKConfiguration.setup&show=true

I personally like to keep an installer around permanently and then the launch button just works, but copy and paste should work too:

![image](https://github.com/eclipse-platform/eclipse.platform/assets/208716/fffa03dc-2e9d-451c-8855-5577c43c5276)

Or maybe dragging the more direct link on the instruction page works.

(What is WSL2?)

**Reply by HeikoKlare** [reply]
WSL is the Windows Subsystem for Linux. Version 2 (WSL2) also has support for GUI applications, which makes it possible to run a Linux Eclipse on a Windows host. I sometimes also use this for simple implementation/validation purposes on Linux (but actually not for critical UI changes, where I prefer a native Linux system).

What I usually do it just to use the Eclipse Installer on WSL2. It does everything you need out of the box (as also described on the setup page linked by Ed).

**Reply by fedejeanne** [reply]
Thank you @merks , that pointed me in the right direction!

Here's what I did (**the order matters**):
1. Clicked on the button in [the documentation](https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#creating-an-eclipse-development-environment)
2. Unfolded the section "Install Using Drag and Drop"
3. Unfolded the section "Install Using Copy and Apply" (BTW @merks , shouldn't this be visible **without** the need to unfold "Install using Drag and Drop"?)
4. Clicked on "Copy Link"
5. Started the Oomph installer in WSL2 (it starts in "Advanced mode" in my case)
6. Clicked the button "Apply the configuration from the clipboard". **This button wasn't visible in my previous screenshot since I hadn't copied the link before opening the installer**

> (What is WSL2?)

As @HeikoKlare mentioned, (very shortly summarized) it's the subsystem one needs in order to run "Linux applications" on Windows. If you have a rather "new" Windows 10 and the necessary drivers for your Graphic Card, you can even start GUI applications (more infos [here](https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps#prerequisites)).

Thank you both for your help!
